pkg/planner/core/indexmerge_path.go

```diff
@@ -376,12 +449,58 @@ func (ds *DataSource) buildIndexMergeOrPath(
 	return indexMergePath
 }
 
+func (ds *DataSource) generateNormalIndexPartialPath4Or(dnfItems []expression.Expression, usedAccessMap []bool, normalPathCnt int) ([]*util.AccessPath, []bool, bool, error) {
+	remainedDNFItems := make([]expression.Expression, 0, len(dnfItems))
+	for i, b := range usedAccessMap {
+		if !b {
+			remainedDNFItems = append(remainedDNFItems, dnfItems[i])
+		}
+	}
+	noMVIndexPartialPath := false
+	if len(dnfItems) == len(remainedDNFItems) {
+		// there is no mv index paths generated, so for: (a<1) OR (a>2), no need to generated index merge.
+		noMVIndexPartialPath = true
+	}
+	paths, needSelection, usedMap, err := ds.generateNormalIndexPartialPaths4DNF(remainedDNFItems, normalPathCnt)
+	if err != nil {
+		return nil, usedAccessMap, false, err
+	}
+	// If all the partialPaths use the same index, we will not use the indexMerge.
+	singlePath := true
+	for i := len(paths) - 1; i >= 1; i-- {
+		if paths[i].Index != paths[i-1].Index {
+			singlePath = false
+			break
+		}
+	}
```

Comment:
I think most of the logic here (including the logic of `generateNormalIndexPartialPaths4DNF`) is the same as `generateIndexMergeOrPaths`, can we try to reuse them?

Reply:
basically and partially can be reused, I can make it in next pr, otherwise, review will be a burden